Frontend/android-native/app/src/main/java/in/juspay/mobility/CommonJsInterface.java

```diff
@@ -2616,47 +2630,80 @@ public void run() {
+            for (int i = 0; i < coor.length(); i++) {
+                try {
+                    JSONArray coordinate = (JSONArray) coor.get(i);
+                    double lat = (double) coordinate.get(1);
+                    double lon = (double) coordinate.get(0);
+                    all_latitudes.add(lat);
+                    all_longitudes.add(lon);
+                } catch (JSONException e) {
+                    e.printStackTrace();
+                }
+            }
+
+            double minLat = Collections.min(all_latitudes);
+            double maxLat = Collections.max(all_latitudes);
+            double minLon = Collections.min(all_longitudes);
+            double maxLon = Collections.max(all_longitudes);
+
+            double left = minLon - 0.1*(maxLon - minLon);
+            double right = maxLon + 0.1*(maxLon - minLon);
+            double top = maxLat + 0.1*(maxLat - minLat);
+            double bottom = minLat - (maxLat - minLat);
+
+            LatLng topLeft = new LatLng(top, left);
+            LatLng bottomRight = new LatLng(bottom, right);
+
+            ArrayList<LatLng> result = new ArrayList<>();
+            result.add(topLeft);
+            result.add(bottomRight);
+
+            return result;
+
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+        return null;
+    }
+
     @JavascriptInterface
     public void drawPolygon(String geoJson, String locationName) throws JSONException {
 
         activity.runOnUiThread(new Runnable() {
             @Override
             public void run() {
-                System.out.println("Inside drawPolygon function");
                 if(layer != null){
                     layer.removeLayerFromMap();
                 }
-                JSONObject geo = null;
-                try {
-                    geo = new JSONObject(geoJson);
-                    PatternItem DASH = new Dash(20);
-                    PatternItem GAP = new Gap(20);
-                    List<PatternItem> PATTERN_POLYGON_ALPHA = Arrays.asList(GAP, DASH);
-                    layer = new GeoJsonLayer(googleMap, geo);
-                    GeoJsonPolygonStyle polyStyle = layer.getDefaultPolygonStyle();
-                    polyStyle.setFillColor(Color.argb(25,0, 102, 255));
-                    polyStyle.setStrokePattern(PATTERN_POLYGON_ALPHA);
-                    polyStyle.setStrokeWidth(2);
-                    polyStyle.setStrokeColor(Color.BLUE);
-                    if (locationName.length() > 0){
-                        zoom = 14.0f;
-                        if (userPositionMarker == null) {
-                            upsertMarker(CURRENT_LOCATION, String.valueOf(getKeyInNativeSharedPrefKeys("LAST_KNOWN_LAT")), String.valueOf( getKeyInNativeSharedPrefKeys("LAST_KNOWN_LON")),160, 0.5f,0.9f); //TODO this function will be removed
-                        } else {
+                if (googleMap != null) {
+                    try {
```

Comment:
function signature already says `throws JSONException`. so try catch block is not reqiured